blog/2025/01-01-efcore-and-me/index.mdx

```diff
@@ -0,0 +1,56 @@
+---
+title: EFCore refactoring is here
+description: EFCore, Jellyfin and me.
+authors: JPVenson
+slug: efcore-refactoring
+tags: [unstable, warning, efcore]
+---
+
+EFCore is here and this will have consequences.
+
+We finally reached our first milestone in cleaning up the legacy code regarding the database access, that means all SQL builders that were targeting SQLite directly are now removed from code.  
+
+This marks the first step towards a completely new database design. But we now need to take a quick look ahead and see what's next.
+
+Please continue to read further to see what exactly that means and what the future brings.
+
+\- JPVenson
+
+{/* truncate */}
+
+## What exactly have we done
+
+[Josh’s blog post](../../2024/11-03-efcore-refactoring-incoming/index.mdx) in November 2024 outlined much of what we aimed to achieve, and I want to take this opportunity to address the feedback and questions we've received, clarifying a few key points.  
+
+The old code included many instances of direct access to the underlying SQLite database.
+SQLite is a file based database where Jellyfin is storing most of its important data.
+This has been an issue for 3 reasons.
+1. The old code was very poorly written
+2. The old code was very complex
+3. It was explicitly written to support SQLite only and nothing else  
+
+Because of this, we were unable to quickly and reliably extend our database. The new code uses EntityFramework migrations, which is the industry standard for database migrations.  
+
+## What haven't we done
+
+While we are currently working on it, the current PR does __not__ add support for alternative database providers on its own. 
+Other database providers will be a significant change in itself, and when introduced, it will likely remain highly experimental for quite some time.  
+
+Very importantly, we also have not yet optimised the rest of Jellyfin to work with the new database access. That means it is quite possible that unstable builds might be significatly slower while we are rewriting the rest of the code to utilise the new structures.  
+
+There are also already plans that build upon the new architecture to provide proper means to back up a Jellyfin instance now that everything is strongly typed. This means a plugin can now back up the entire Jellyfin instance while it is running, something that was previously not possible to do reliably.
+Furthermore, this export/import plugin would be necessary if we aim to support multiple database providers and allow exporting data to other databases.
```

Comment:
```suggestion
Furthermore, this export/import plugin would be necessary if we aim to support multiple database providers and allow exporting data to other databases systems.
```

Reply:
Drop the "s" after databases - database systems - but yes.